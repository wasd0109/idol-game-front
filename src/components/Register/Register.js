import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import ErrorBar from '../ErrorBar/index';

function Register({ onRegister, error, resetError }) {
  const [email, setEmail] = useState('');
  const [playerName, setPlayerName] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  return (
    <div className="flex justify-center mt-4 md:mt-16">
      <div className="w-full max-w-xs">
        <form className="bg-white shadow-md rounded px-8 pt-6 pb-2 mb-4">
          <div className="mb-4">
            <label
              className="block text-gray-700 text-sm font-bold mb-2"
              htmlFor="playername"
            >
              Player Name
            </label>
            <input
              className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
              id="playername"
              type="text"
              placeholder="Player Name"
              value={playerName}
              onChange={(event) => setPlayerName(event.target.value)}
            />
          </div>
          <div className="mb-4">
            <label
              className="block text-gray-700 text-sm font-bold mb-2"
              htmlFor="email"
            >
              Email
            </label>
            <input
              className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
              id="email"
              type="text"
              placeholder="email"
              value={email}
              onChange={(event) => setEmail(event.target.value)}
            />
          </div>
          <div className="mb-4">
            <label
              className="block text-gray-700 text-sm font-bold mb-2 "
              htmlFor="password"
            >
              Password
            </label>
            <input
              className="shadow appearance-none border  rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
              id="password"
              type="password"
              placeholder="******************"
              value={password}
              onChange={(event) => setPassword(event.target.value)}
            />
          </div>
          <div className="mb-4">
            <label
              className="block text-gray-700 text-sm font-bold mb-2 "
              htmlFor="password"
            >
              Confirm Password
            </label>
            <input
              className="shadow appearance-none border  rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
              id="confirmPassword"
              type="password"
              placeholder="******************"
              value={confirmPassword}
              onChange={(event) => setConfirmPassword(event.target.value)}
              role="ConfirmPassword"
            />
          </div>
          <div className="flex items-center justify-between mt-4">
            <Link
              className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
              type="submit"
              onClick={(event) => {
                event.preventDefault();
                resetError();
                onRegister(email, password, confirmPassword, playerName);
              }}
              to="/dashboard"
            >
              Register
            </Link>
            <Link
              to="/login"
              className="inline-block align-baseline font-bold text-sm text-blue-500 hover:text-blue-800"
              onClick={resetError}
            >
              <p>Login</p>
            </Link>
          </div>
          <div className="mt-4 mb-2">
            {error ? <ErrorBar msg={error} /> : null}
          </div>
        </form>
        <div className="text-center">
          <a
            rel="noopener noreferrer"
            target="_blank"
            href="https://myidol46.netlify.app/privacy"
          >
            Privacy Claim
          </a>
        </div>
      </div>
    </div>
  );
}

export default Register;
