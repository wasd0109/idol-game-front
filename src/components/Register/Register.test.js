import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import Register from './Register';

let component;

const initialProps = {
  onRegister: jest.fn(),
  error: '',
  isLoggingIn: false,
  resetError: jest.fn(),
};

beforeEach(() => {
  component = render(
    <Router>
      <Register {...initialProps} />
    </Router>
  );
});

describe('Register component render correctly', () => {
  test('Register component render', () => {
    expect(component).toMatchSnapshot();
  });

  test('Register render error correctly', () => {
    const props = {
      onSubmit: jest.fn(),
      error: 'Error',
      isLoggingIn: false,
      resetError: jest.fn(),
    };
    const { getByText } = render(
      <Router>
        <Register {...props} />
      </Router>
    );
    expect(getByText(props.error)).toBeInTheDocument();
  });
});

describe('Register component function correctly', () => {
  test('Link to login functional', () => {
    userEvent.click(screen.getByText('Login'));
    expect(initialProps.resetError).toBeCalled();
    expect(document.URL).toEqual('http://localhost/login');
  });

  test('User input handled correctly', async () => {
    const playerName = 'Testuser123';
    const email = 'Testuser123@gmail.com';
    const password = 'testuser123';
    const confirmPassword = 'testuser123';
    await userEvent.type(screen.getByText('Player Name'), playerName);
    await userEvent.type(screen.getByLabelText('Email'), email);
    await userEvent.type(screen.getByLabelText('Password'), password);
    await userEvent.type(screen.getByRole('ConfirmPassword'), confirmPassword);
    expect(screen.getByLabelText('Player Name').value).toEqual(playerName);
    expect(screen.getByLabelText('Email').value).toEqual(email);
    expect(screen.getByLabelText('Password').value).toEqual(password);
    expect(screen.getByLabelText('Confirm Password').value).toEqual(
      confirmPassword
    );
  });

  test('Register form submit with correct information', async () => {
    expect.assertions(1);
    const playerName = 'Testuser123';
    const email = 'Testuser123@gmail.com';
    const password = 'testuser123';
    const confirmPassword = 'testuser123';
    await userEvent.type(screen.getByText('Player Name'), playerName);
    await userEvent.type(screen.getByLabelText('Email'), email);
    await userEvent.type(screen.getByLabelText('Password'), password);
    await userEvent.type(screen.getByRole('ConfirmPassword'), confirmPassword);
    userEvent.click(screen.getByText('Register'));
    expect(initialProps.onRegister).toBeCalledWith(
      email,
      password,
      confirmPassword,
      playerName
    );
  });
  test('Register handle incorrect confirmed password properly', async () => {});
});
