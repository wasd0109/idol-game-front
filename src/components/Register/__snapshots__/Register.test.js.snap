// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component render correctly Register component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-center mt-4 md:mt-16"
      >
        <div
          class="w-full max-w-xs"
        >
          <form
            class="bg-white shadow-md rounded px-8 pt-6 pb-2 mb-4"
          >
            <div
              class="mb-4"
            >
              <label
                class="block text-gray-700 text-sm font-bold mb-2"
                for="playername"
              >
                Player Name
              </label>
              <input
                class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
                id="playername"
                placeholder="Player Name"
                type="text"
                value=""
              />
            </div>
            <div
              class="mb-4"
            >
              <label
                class="block text-gray-700 text-sm font-bold mb-2"
                for="email"
              >
                Email
              </label>
              <input
                class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
                id="email"
                placeholder="email"
                type="text"
                value=""
              />
            </div>
            <div
              class="mb-4"
            >
              <label
                class="block text-gray-700 text-sm font-bold mb-2 "
                for="password"
              >
                Password
              </label>
              <input
                class="shadow appearance-none border  rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
                id="password"
                placeholder="******************"
                type="password"
                value=""
              />
            </div>
            <div
              class="mb-4"
            >
              <label
                class="block text-gray-700 text-sm font-bold mb-2 "
                for="password"
              >
                Confirm Password
              </label>
              <input
                class="shadow appearance-none border  rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
                id="confirmPassword"
                placeholder="******************"
                role="ConfirmPassword"
                type="password"
                value=""
              />
            </div>
            <div
              class="flex items-center justify-between mt-4"
            >
              <button
                class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
                type="submit"
              >
                Register
              </button>
              <a
                class="inline-block align-baseline font-bold text-sm text-blue-500 hover:text-blue-800"
                href="/login"
              >
                <p>
                  Login
                </p>
              </a>
            </div>
            <div
              class="mt-4 mb-2"
            />
          </form>
          <div
            class="text-center"
          >
            <a
              href="https://myidol46.netlify.app/privacy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy Claim
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-center mt-4 md:mt-16"
    >
      <div
        class="w-full max-w-xs"
      >
        <form
          class="bg-white shadow-md rounded px-8 pt-6 pb-2 mb-4"
        >
          <div
            class="mb-4"
          >
            <label
              class="block text-gray-700 text-sm font-bold mb-2"
              for="playername"
            >
              Player Name
            </label>
            <input
              class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
              id="playername"
              placeholder="Player Name"
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="block text-gray-700 text-sm font-bold mb-2"
              for="email"
            >
              Email
            </label>
            <input
              class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
              id="email"
              placeholder="email"
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="block text-gray-700 text-sm font-bold mb-2 "
              for="password"
            >
              Password
            </label>
            <input
              class="shadow appearance-none border  rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
              id="password"
              placeholder="******************"
              type="password"
              value=""
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="block text-gray-700 text-sm font-bold mb-2 "
              for="password"
            >
              Confirm Password
            </label>
            <input
              class="shadow appearance-none border  rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
              id="confirmPassword"
              placeholder="******************"
              role="ConfirmPassword"
              type="password"
              value=""
            />
          </div>
          <div
            class="flex items-center justify-between mt-4"
          >
            <button
              class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
              type="submit"
            >
              Register
            </button>
            <a
              class="inline-block align-baseline font-bold text-sm text-blue-500 hover:text-blue-800"
              href="/login"
            >
              <p>
                Login
              </p>
            </a>
          </div>
          <div
            class="mt-4 mb-2"
          />
        </form>
        <div
          class="text-center"
        >
          <a
            href="https://myidol46.netlify.app/privacy"
            rel="noopener noreferrer"
            target="_blank"
          >
            Privacy Claim
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
